lib/rules/no-decorators-before-export.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+/** @type {import('eslint').Rule.RuleModule} */
+
+function reportInvalidSyntax(context, node) {
+  if (isInvalidSyntax(node)) {
+    const isDefaultExport = node.type === 'ExportDefaultDeclaration';
+    const isNamed = node.declaration.id !== null;
+    const name = isNamed ? node.declaration.id.name : '';
+    const numDecorators = node.declaration.decorators.length;
+
+    const message = `Usage of class decorator${numDecorators > 1 ? 's' : ''} on the ${
+      isNamed ? '' : 'un-named '
+    }${isDefaultExport ? 'default ' : ''}export${
+      isNamed ? ` ${name}` : ''
+    } must occur prior to exporting the class.`;
+
+    if (isNamed) {
+      const sourceCode = context.getSourceCode();
+      const src = sourceCode.getText(node);
+      context.report({
+        node,
+        message,
+        fix(fixer) {
+          if (isDefaultExport) {
+            let newSrc = src.replace(`export default class ${name}`, `class ${name}`);
+            newSrc = `${newSrc}\nexport default ${name};\n`;
+            return fixer.replaceText(node, newSrc);
+          }
+          let newSrc = src.replace(`export class ${name}`, `class ${name}`);
+          newSrc = `${newSrc}\nexport { ${name} };\n`;
+          return fixer.replaceText(node, newSrc);
+        },
+      });
+    } else {
+      context.report({ node, message });
+    }
+  }
+}
+
+function isInvalidSyntax(node) {
+  return (
+    node.declaration &&
+    node.declaration.type === 'ClassDeclaration' &&
+    node.declaration.decorators?.length
+  );
+}
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description:
+        'disallow the use of a class decorator on an export declaration. Enforces use of the decorator on the class directly before exporting the result.',
+      category: 'Miscellaneous',
+      recommended: true,
```

Comment:
Note: recommended rules are only added in major versions should this should be removed.